src/cell.ts

```diff
@@ -0,0 +1,52 @@
+import { CellId, CellNick } from '@holochain/conductor-api'
+import { fakeCapSecret } from './common'
+import { Player } from './player'
+
+type CallZomeFunc = (zome: string, fn: string, params?: any) => Promise<any>
+
+type CellConstructorParams = {
+  cellId: CellId,
+  cellNick: CellNick,
+  player: Player
+}
+
+/**
+ * Make zome calls from Cells.
+ * `cell.call('zomeName', 'funcName', params)`
+ */
+export class Cell {
+  cellId: CellId
+  cellNick: CellNick
+  _player: Player
+
+  constructor(o: CellConstructorParams) {
+    this.cellId = o.cellId
+    this.cellNick = o.cellNick
+    this._player = o.player
+  }
+
+  call: CallZomeFunc = async (zome: string, fn: string, params?: any): Promise<any> => {
+    if (typeof zome !== 'string' || typeof fn !== 'string') {
+      throw new Error("cell.call() must take at least `zome` and `fn` args")
+    }
+
+    // FIXME: don't just use provenance from CellId that we're calling,
+    //        (because this always grants Authorship)
+    //        for now, it makes sense to use the AgentPubKey of the *caller*,
+    //        but in the future, Holochain will inject the provenance itself
+    //        and you won't even be able to pass it in here.
+    const [_dnaHash, provenance] = this.cellId
+    return this._player.app(`cell.call()`).callZome({
```

Comment:
@zippy 
This is a big improvement here now: having the _player reference and calling `.app(...)` invokes the `_conductorGuard` of the player, meaning if it has gone into shutdown it will `throw` which is what we want.

Reply:
This was confusing me at first because I was imagining this was getting a reference to an app with a certain id, rather than access to the Maybe Interface. It's probably fine. Maybe though, you could access these through `player.client.app()` and `player.client.admin()` instead.

Reply:
yeah.... ugh, I didn't like the naming, but Eric called the one `.admin()` and then I needed to add this and thought it'd be weird if it didn't mirror the other one. 
Sadly though I knew that Eric and co have added code in their rush to update to this branch that uses the `.admin()` call so I was avoiding renaming it. 
I'd like to rename it though so I propose they should update because this ain't quite right

Reply:
renamed adminWs() and appWs()